File: python/7 kyu/339.py
# https://www.codewars.com/kata/5aa39ba75084d7cf45000008/train/python

# Passed

def solve(n, memo=dict()):
    if (n in memo):
        result = memo[n]
        return result
    
    if (n == 0):
        return "0"
    elif (n == 1):
        return "01"
    
    result = solve(n - 1) + solve(n - 2)
    memo[n] = result
    return result

output = solve(3)
print(output)

File: nodejs/8 kyu/339.js
// https://www.codewars.com/kata/56b29582461215098d00000f/train/javascript

// Passed

function pipeFix(numbers) {
  let minNumber = numbers[0];
  let maxNumber = numbers[numbers.length - 1];
  let result = [];
  for (let num = minNumber; num <= maxNumber; num++) {
    result.push(num);
  }

  return result;
}

const output = pipeFix([1, 2, 3, 12]);
console.log(output);